catalog/app/model/S3.ts

```diff
@@ -0,0 +1,5 @@
+export interface S3ObjectLocation {
```

Comment:
We already have `S3HandleBase`. If they meant the same, I think we should leave either this, or S3HandleBase.

As a starting point, we can change `S3HandleBase`:

```ts
import type { S3ObjectLocation } from 'model'
type S3HandleBase = S3ObjectLocation
```

Reply:
more or less the same, yeah -- i'm trying to introduce a cleaner and stricter abstraction (S3HandleBase is used in too many different and sometimes incompatible contexts). migrating and unifying all the s3handle-based use-cases is out of scope of this PR tho